txnbuild/transaction.go

```
@@ -851,11 +846,8 @@ func ReadChallengeTx(challengeTx, serverAccountID, network string) (tx *Transact
 		return tx, clientAccountID, errors.New("challenge cannot be a fee bump transaction")
 	}
 
-	// Enforce no muxed accounts (at least until we understand their impact)
```

I think we should still have this check because we do not expect anyone to use a muxed account here and I think we should return an error in this case

---

I will see what I can do without using strkeys.

---

Done